server/data/data-postgres/src/main/java/org/apache/james/sieve/postgres/PostgresSieveScriptDAO.java

```diff
@@ -39,20 +39,43 @@
 import org.apache.james.sieve.postgres.model.PostgresSieveScript;
 import org.apache.james.sieverepository.api.ScriptName;
 import org.jooq.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import io.r2dbc.spi.Result;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class PostgresSieveScriptDAO {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresSieveScriptDAO.class);
+
+    public static void enforceMaximumOneActiveScriptPerUserConstraint(PostgresExecutor postgresExecutor) {
+        postgresExecutor.connection()
+            .flatMapMany(connection -> connection.createStatement(String.format("ALTER TABLE %s ADD CONSTRAINT maximum_one_active_script_per_user EXCLUDE (%s WITH =) WHERE (%s = TRUE);",
+                    TABLE_NAME.getName(), USERNAME.getName(), IS_ACTIVE.getName()))
+                .execute())
+            .flatMap(Result::getRowsUpdated)
+            .then()
+            .onErrorResume(e -> {
+                if (e.getMessage().contains("already exists")) {
+                    return Mono.empty();
+                }
+                LOGGER.error("Error while creating constraint for table {}", TABLE_NAME.getName(), e);
+                return Mono.error(e);
+            })
+            .block();
+    }
+
     private final PostgresExecutor postgresExecutor;
 
     @Inject
     public PostgresSieveScriptDAO(@Named(DEFAULT_INJECT) PostgresExecutor postgresExecutor) {
         this.postgresExecutor = postgresExecutor;
+        enforceMaximumOneActiveScriptPerUserConstraint(postgresExecutor);
```

Comment:
Please find a way to do this at table creation

Reply:
The rationale behind that is jooq does not have DSL support for EXCLUDE constraint (only for primary key, foreign key and unique constraints), and no raw SQL to create constraint at the same time table creation.

Reply:
Maybe I would think about adding raw sql query in the `PostgresModule`:
```
    List<String> additionalConstraints();
```

and upon `PostgresTableManager` initiation, create the constraints using the raw query too.

Reply:
I mean: alter statement is OK but find a way to attach it to the table declaration and execute it just after table creation.

Reply:
> and upon PostgresTableManager initiation, create the constraints using the raw query too.

Yes that's what I want!